mne/source_estimate.py

```diff
@@ -272,13 +274,15 @@ def read_source_estimate(fname, subject=None):
                      for f in [fname + '-rh.stc', fname + '-lh.stc']]
         w_exist = [op.exists(f)
                    for f in [fname + '-rh.w', fname + '-lh.w']]
-        h5_exist = op.exists(fname + '-stc.h5')
         if all(stc_exist) and (ftype is not 'w'):
             ftype = 'surface'
         elif all(w_exist):
             ftype = 'w'
-        elif h5_exist:
+        elif op.exists(fname + '.h5'):
             ftype = 'h5'
+        elif op.exists(fname + '-stc.h5'):
+            ftype = 'h5'
+            fname += '-stc'
```

Comment:
I don't understand. `fname` is used below a lot. And no, 320 should definitely be `elif` as far as I can make out. Granted, the logic in this function is horribly complicated. Any suggestions for improving clarity here are appreciated, but should probably go into a separate PR.